### ISSUE TITLE
why the published post doesn't appear in the homepage?

### ISSUE THREAD

**Issue by luckypoem** [zserge/zserge.github.io#4]
hi.
 
as per ur repo,i setup a blog: http://zs.bright.biz.st/ ,but a published post: http://zs.bright.biz.st/blog/test-1.html doesn't appear in the homepage.
why?how to fix it?
tks

**Comment by zserge** [comment]
I normally just call `zs build` from the src directory in my blog and it generates RSS XML, as well as the front page with the list of recent posts. The script that builds the front page is here - https://github.com/zserge/zserge.github.io/blob/master/src/.zs/blogposts (as long as you have bash and basic unix tools - it should work fine). After that I publish all files from my `.pub` subdirectory, including the generated `blog.html`

**Comment by luckypoem** [comment]
hi.

root@localhost:/usr/local/zserge.github.io/src# ls .pub
404.html    blog       CNAME        images      jsmn.html      rss.xml
about.html  blog.html  favicon.ico  index.html  projects.html  styles.css
root@localhost:/usr/local/zserge.github.io/src# ls .zs
blogposts    ga.html  layout.amber  preview      webfont.html
disqus.html  genrss   prehook       social.html
root@localhost:/usr/local/zserge.github.io/src# .zs/blogposts
<ul>
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
.zs/blogposts: 11: .zs/blogposts: var: not found
</ul>
root@localhost:/usr/local/zserge.github.io/src# 

what does the export generated by ".zs/blogposts" mean?

**Comment by luckypoem** [comment]
@zserge 
could u kindly reply me?

**Comment by zserge** [comment]
Not sure why you're trying to execute `.zs/blogposts` script directly. It's meant to be called by `zs` binary. The whole blog is "built" using the `zs` tool (https://github.com/zserge/zs). Once you get `zs` installed - simply do `zs build` from your source directory and it should generate the rss feed and the blog index page for you.

**Comment by luckypoem** [comment]
@zserge 

root@localhost:/usr/local/zserge.github.io/src# zs build
it shows:
...
2017/04/11 15:04:21 build: blog/anvil-app-architecture.md
2017/04/11 15:04:21 build: blog/anvil-breaking-changes.md
2017/04/11 15:04:21 build: blog/anvil-forge-merged.md
2017/04/11 15:04:21 build: blog/anvil-howto-style-views.md
2017/04/11 15:04:21 build: blog/anvil-kotlin.md
2017/04/11 15:04:21 build: blog/anvil-support-libraries.md
2017/04/11 15:04:21 build: blog/anvil-two-way-bindings.md
2017/04/11 15:04:21 build: blog/anvil-xml-support.md
2017/04/11 15:04:21 build: blog/buck-build-system.md
2017/04/11 15:04:21 build: blog/cucu-part1.md
2017/04/11 15:04:21 build: blog/cucu-part2.md
2017/04/11 15:04:21 build: blog/cucu-part3.md
2017/04/11 15:04:21 build: blog/go-cloud-benchmark.md
2017/04/11 15:04:21 build: blog/gradle-maven-publish.md
2017/04/11 15:04:21 build: blog/hello.md
2017/04/11 15:04:21 build: blog/jitpack.md
2017/04/11 15:04:21 build: blog/kotlin.md
2017/04/11 15:04:21 build: blog/log.md
2017/04/11 15:04:21 build: blog/logdog.md
2017/04/11 15:04:21 build: blog/luash.md
2017/04/11 15:04:21 build: blog/minimal-testing.md
2017/04/11 15:04:21 build: blog/mucks.md
2017/04/11 15:04:21 build: blog/mucks2.md
2017/04/11 15:04:21 build: blog/my-experience-with-asmjs.md
2017/04/11 15:04:21 build: blog/new-site-generator.md
2017/04/11 15:04:21 build: blog/new-site-generator2.md
2017/04/11 15:04:21 build: blog/palette.md
2017/04/11 15:04:21 build: blog/tcl-interpreter.md
2017/04/11 15:04:21 build: blog/test-1.md
2017/04/11 15:04:21 build: blog/update-jknav.md
2017/04/11 15:04:21 build: blog.md
2017/04/11 15:04:22 ERROR: .zs/blogposts: 12: [: 2017-03-27: unexpected operator

2017/04/11 15:04:22 build: favicon.ico
2017/04/11 15:04:22 build: images/issue-tracker.png
2017/04/11 15:04:22 build: images/logdog.png
2017/04/11 15:04:22 build: images/mucks1.png
2017/04/11 15:04:22 build: images/mucks2.png
2017/04/11 15:04:22 build: images/mucks3.png
2017/04/11 15:04:22 build: images/mucks4.png
2017/04/11 15:04:22 build: images/mucks5.png
2017/04/11 15:04:22 build: images/six.png
2017/04/11 15:04:22 build: images/textizer-calendar-1x1.png
2017/04/11 15:04:22 build: images/textizer-widget-4x1.png
2017/04/11 15:04:22 build: index.md
2017/04/11 15:04:22 ERROR: .zs/preview: 5: [: 2017-03-27: unexpected operator

2017/04/11 15:04:22 build: jsmn.md
2017/04/11 15:04:22 build: projects.amber
2017/04/11 15:04:22 build: rss.xml
2017/04/11 15:04:22 ERROR: .zs/genrss: 9: [: 2017-03-27: unexpected operator

2017/04/11 15:04:22 build: styles.gcss
root@localhost:/usr/local/zserge.github.io/src#

how to fix the error: "2017/04/11 15:04:22 ERROR: .zs/blogposts: 12: [: 2017-03-27: unexpected operator"?
thanks.

**Comment by luckypoem** [comment]
Can u kindly pls reply?
Tks

**Comment by zserge** [comment]
What kind of shell do you use? Are you familiar with shell scripting?
For some reason `if [ ! -z $d ] ; then` is not a valid expression in your shell.
You may try replacing that with `if [ "x$d" != "x" ] ; then` and see if that helps.

**Comment by zserge** [comment]
Or it may be a more typical case when people use improper text editors. I haven't got any "2017-03-27" dates in my blog, so that must be from the file you've created. If you're using windows and put windows-style line endings - most unix shell scripts will break on it. You may try converting the line endings to UNIX (`\n`) in your text editor and try once again.

**Comment by luckypoemster** [comment]
i know why now.about the value of date: after 2017-03-27,i added "14:20:00" ,but actually the part of "14:20:00" isn't supported.

**Comment by zserge** [comment]
Correct, date is expected to be only the _date_, and it should be in the format YYYY-MM-DD.
